app/models/activity.rb

```diff
@@ -26,6 +26,10 @@ def project=(new_project); end
   scope :latest, -> do
     includes(:trackable).order('activities.created_at DESC').limit(20)
   end
+  scope :by_user, -> (id) { where user_id: id }
+  scope :by_type, -> (type) { where trackable_type: type }
+  scope :by_period_start, -> (period_start) { where('created_at >= ?', period_start) }
+  scope :by_period_end, -> (period_end) { where('created_at <= ?', period_end) }
```

Comment:
Unlike the 2 lines above, I can get behind this syntactic sugar. However, `Activity.by_period_start` makes me think that we're searching for activities in that exact date, so maybe we can improve the semantics? Check out this method that does the exact same thing: https://github.com/dradis/dradis-ce/blob/main/app/models/notification.rb#L21. `Activity.since(Date.yesterday)` reads better in my opinion. On the other hand, `Activity.before(Date.today)` sounds better than `Activity.by_period_end(Date.today)`

Reply:
I like this. changing it!